### DISCUSSION TITLE
How to fix POSH broken prompt on vscode?

### DISCUSSION THREAD

**Discussion by prathikshetty2002** [JanDeDobbeleer/oh-my-posh#2089] [Troubleshoot] [answered]
I recently started using ohmyposh with my windows powershell prompt. It looks great on my terminal. But as soon as i switch to my vscode to do my development stuff, it seems to be broken. It looks bad , i want to fix it. Any suggestions/solutions on it? Have you guys also faced this before. 

This is my powershell with ohmyposh tweaks :

![Screenshot 2022-04-15 000030](https://user-images.githubusercontent.com/63772910/163454256-6a903fd0-c74a-4297-84cf-ce27b13d1b8d.png)

This is how it looks on vscode on windows:

![Screenshot 2022-04-15 000128](https://user-images.githubusercontent.com/63772910/163454384-8f324020-165b-4790-90ea-831792213c5f.png)

**Comment by JanDeDobbeleer** [comment]
Can it be it starts on a UNC path? What's the value of `$PWD`?

**Reply by prathikshetty2002** [reply]
It was displaying broken unicode on the prompt before but a couple of days after i encountered this problem. Does it have to do anything with my powershell version? I am on powershell 5.1 now, they have an upgrade to 7.

**Reply by prathikshetty2002** [reply]
Here is my $PROFILE script :

Import-Module oh-my-posh
oh-my-posh prompt init pwsh --config https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/v$(oh-my-posh --version)/themes/mt.omp.json | Invoke-Expression
Set-PSReadLineOption -PredictionSource History
Set-PSReadLineOption -PredictionViewStyle ListView

maybe you can find something in this. Any issue here?

**Reply by JanDeDobbeleer** [reply]
Why does the path have two slashes `C://`?

**Reply by prathikshetty2002** [reply]
oh sry abt that 

my actual path : C:\Users\prath

**Comment by prathikshetty2002** [comment]
I really find it strange but it actually now reverted back to the previous state after a restart. Don't know how but yes it did!

Still broken unicode prompt :

![Screenshot 2022-04-15 010354](https://user-images.githubusercontent.com/63772910/163463291-449279bd-ee12-470e-b939-ab52c33a50ee.png)

Can't display those nice little icons, it distracts me any fixes you can suggest? I really like the ohmyposh shell !

**Reply by JanDeDobbeleer** [reply]
That's a font issue. All you need is to set the [terminal font in code](https://code.visualstudio.com/docs/editor/integrated-terminal#_terminal-appearance) to one that supports powerline (preferably a [Nerd Font](https://www.nerdfonts.com/)).

**Reply by prathikshetty2002** [reply]
I did that i have set the terminal font to cascadia-cove nerd font and it doesn't supports powerline . I found out another nerd font Fira code that does support powerline fonts. Thanks @JanDeDobbeleer this resolves my issue, if something still persists i will get back for help. Thanks !

**Reply by MeetBhingradiya** [reply]
i Totally Fixed this  issue in ```Settings.Json```

if you install oh-my-posh terminal theme with nerd fonts.

Simply Add this line
```JSON
"terminal.integrated.gpuAcceleration": "auto",
"terminal.integrated.fontFamily": "'CaskaydiaCove Nerd Font Mono'",
```

@prathikshetty2002

**Comment by prathikshetty2002** [comment, marked as answer]
Changing the font does the work , NERD font fira code works well. Some tweaks fixed the issue. Thanks !

**Comment by benn511** [comment]
My mistake was that I had previously set my terminal font and it was overriding the editor font family setting. 
  "terminal.integrated.fontFamily": "Hack Nerd Font"

**Comment by sadranyi** [comment]
I have an interesting issue, all my icons and glyphs work fin in the external terminal normally when I run it in both normal and elevated modes (as seen from the attached image). However, when launching VSCode, it's missing some of the icons and glyphs though I have installed Melso Nerd font and set the Editor Font, which is the default for the Terminal. Kindly advise? 

![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/564700/2235d238-5c24-46a6-848d-6dffcc4916d0)

![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/564700/54623ad5-885e-42a9-9093-63cbb44b2fc7)

![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/564700/ccd07923-1b83-446e-be52-aa66cda77900)

![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/564700/af81a8d1-21b2-4487-bba0-2171cb8ceb19)

![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/564700/d7360e41-e08a-4469-b225-0dabf8bb911d)

As you can see in the VSCode Terminal, 3 of the glyphs are missing.

**Reply by JanDeDobbeleer** [reply]
@sadranyi It's not going to be using the same font. Otherwise vscode has a bug.

**Reply by MeetBhingradiya** [reply]
@sadranyi  Last Image is VS Code Default Font Family Settings `Not` Terminal Font Family See Heare as ScreenShot.

![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/102130001/f78333c2-4428-4226-a1d8-8a8fec46154a)

**Reply by sadranyi** [reply]
I made the changes, yet the icons are missing 
![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/564700/64e3c94b-9e90-4766-b61d-a7210f82c7c1)

**Reply by JanDeDobbeleer** [reply]
@sadranyi is this the latest version of the Nerd Font (V3) and, did you install the font as admin or user?

**Reply by sadranyi** [reply]
yes its the latest, I just selected all and installed them

**Reply by MeetBhingradiya** [reply]
@sadranyi [See Below Comment](https://github.com/JanDeDobbeleer/oh-my-posh/discussions/2089#discussioncomment-6151820)

**Comment by MeetBhingradiya** [comment]
# oh-my-posh Full Installation with Autocompletions and Fix VSCode Broken Issue
>### Note 
>+  I Recommend You only Install `CaskaydiaCove Nerd Font` other forts may be not work.

## Download and Install [Windows 11 as Base in All Paths]
1) Install this App from Microsoft Store is More Recommended : [Link](https://apps.microsoft.com/store/detail/XP8K0HKJFRXGCK)
2) Install `Powershell` from Microsoft Store : [Link](https://www.microsoft.com/store/productId/9MZ1SNWT0N5D)
3) Download `CaskaydiaCove Nerd Font` : [Link](https://www.nerdfonts.com/font-downloads)
4) Unzip `CaskaydiaCove Nerd Font`  it.
5) Go to `Settings > Personalisation > Fonts`  and Then Install it.
6) Search for `Terminal` Run with Administrator.
7) Go to  `Terminal > Settings > Profiles > Defaults > Additional Settings > Appearance > Font Face` Select it to `CaskaydiaCove NFM`
8) Click on Save
9) Go to File Explorer `Documents > PowerShell` create new File `Microsoft.PowerShell_profile.ps1`

> + For No Need of `Autocompletions`
>+ `oh-my-posh --init --shell pwsh --config "D:\Programs\Powershell\config.json" | Invoke-Expression`
>
> + For Need of `Autocompletions` Add Content of This File [Github Gists Link](https://gist.github.com/MeetBhingradiya/9b61d0b16a8cc42f38e8cebef3c072c4)
> + Note : This file is i download from `JanDeDobbeleer's Gists` but now this is removed

10) If You need to Customise or Edit your Theme create config.js file at `Step 9 Path`

## For only Who Need Autocompletions
11) Open `Powershell 7` and Run This command One by One to Install Recomended Modules
+ `Install-Module PSReadLine`
+ `Install-Module Terminal-Icons`
+ `Install-Module z`

12) After a Install you see `Modules` Folder Created in  `Documents > PowerShell`
> ![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/102130001/fb55956b-8689-45ad-82ee-c68710eda524)

## For only Who Need to Fix Broken Powershell in VSCode
+ Press `CTRL + SHIFT + P` and Search for Below Settings Visible and click it
![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/102130001/08673e76-2059-40fb-8bfb-9b2ecd2b05f4)

+ And Added This Properties 
```JSON
// ? Required
"terminal.integrated.fontFamily": "CaskaydiaCove NFM",

// ? Optional For Your Screen And Size Aujest it Manually
"terminal.integrated.fontSize": 12,
"terminal.integrated.fontWeight": "400",
```

## Final Results
![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/102130001/bf22f06f-f03a-4045-adb0-1eb2d0c9827c)

Made With ❤️ by @MeetBhingradiya

**Comment by sadranyi** [comment]
Changing the font to CaskaydiaCove Nerd Font, as suggested by @MeetBhingradiya worked. thanks @MeetBhingradiya. I am still having to run VSCode as Administrator before it picks up the Path Variables. Any pointers?

**Comment by mhudn** [comment]
Using MesloLGLDZ Nerd Font works well for me.
`"terminal.integrated.fontFamily": "MesloLGLDZ Nerd Font",`
![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/32351992/8c93fa68-f74c-44a8-a077-90e134d0cae4)

**Comment by arif98741** [comment]
Its worked in my powershell but when i try to open inside vscode it was not showing. That's why I  open vscode user settings file from location 
`C:\Users\YourPCUserName\AppData\Roaming\Code\User\settings.json` and added a new line at the end like below 

![Screenshot_68](https://github.com/JanDeDobbeleer/oh-my-posh/assets/17213478/d13c9be6-8280-4f16-bb01-bd3726d46128)

` "terminal.integrated.fontFamily": "Hack Nerd Font",`

Finally It's now working 
![Screenshot_69](https://github.com/JanDeDobbeleer/oh-my-posh/assets/17213478/d399a818-24f2-4051-aa56-c4f392e27a72)

**Comment by MeetBhingradiya** [comment]
@JanDeDobbeleer can you lock this conversation so unnecessary comments can't confuse other users.
